homeassistant/components/zwave_js/fan.py

```diff
@@ -86,11 +86,17 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     @property
     def is_on(self) -> bool:
         """Return true if device is on (speed above 0)."""
+        if self.info.primary_value.value is None:
+            # guard missing value
+            return False
```

Comment:
A fan could be on even if we don't know that it is.

Reply:
Our ToggleEntity implementation isn't smart enough yet to distinguish None from False. So it doesn't matter yet, but we can return None to be more correct.

https://github.com/home-assistant/core/blob/a74ae3585a40dd89c38a830dd030e9694094e685/homeassistant/helpers/entity.py#L676

Reply:
Yeah, but I have no idea how to handle that if the value is None/null/undefined.
@MartinHjelmare if I return None here, will HA report is at "unknown" or does it need a boolean value here ?

Reply:
As seen in the link above, we can return None, but the result will be the same. But I'd still prefer that since it's more correct and it will take advantage of a future improvement in the parent class.